scripts/main/photo.js

```diff
@@ -139,33 +139,75 @@ photo.update_display_overlay = function () {
 	}
 };
 
-// Preload the next photo for better response time
-photo.preloadNext = function(photoID) {
+// Preload the next and previous photos for better response time
+photo.preloadNextPrev = function(photoID) {
 	if (album.json &&
 		album.json.photos &&
-		album.getByID(photoID) &&
-		album.getByID(photoID).nextPhoto!=='') {
-
-		let nextPhoto = album.getByID(photoID).nextPhoto;
-		let url       = album.getByID(nextPhoto).url;
-		let medium    = album.getByID(nextPhoto).medium;
-		let href      = url;
-		if (medium != null && medium !== '') {
-			href = medium;
-
-			let medium2x = album.getByID(nextPhoto).medium2x;
-			if (medium2x && medium2x !== '') {
-				// If the currently displayed image uses the 2x variant,
-				// chances are that so will the next one.
-				let imgs=$('img#image');
-				if (imgs.length > 0 && imgs[0].currentSrc != null && imgs[0].currentSrc.includes('@2x.')) {
-					href = medium2x;
+		album.getByID(photoID)) {
+
+		let previousPhotoID = album.getByID(photoID).previousPhoto;
+		let nextPhotoID = album.getByID(photoID).nextPhoto;
+		let current2x = null;
+
+		$('head [data-prefetch]').remove();
+
+		let preload = function(preloadID) {
+			let preloadPhoto  = album.getByID(preloadID);
+			let href = '';
+
+			if (preloadPhoto.medium != null && preloadPhoto.medium !== '') {
+				href = preloadPhoto.medium;
+
+				if (preloadPhoto.medium2x && preloadPhoto.medium2x !== '') {
+					if (current2x === null) {
+						let imgs = $('img#image');
+						current2x = imgs.length > 0 && imgs[0].currentSrc !== null && imgs[0].currentSrc.includes('@2x.')
+					}
+					if (current2x) {
+						// If the currently displayed image uses the 2x variant,
+						// chances are that so will the next one.
+						href = preloadPhoto.medium2x
```

Comment:
Yes and no. I think you still need to check if medium2x is not empty.

Reply:
Is that not covered on line 161?

Reply:
Ho right. xD